src/mainwindow.cpp

```diff
@@ -398,3 +488,80 @@ void MainWindow::aboutHotspot()
     dialog.adjustSize();
     dialog.exec();
 }
+
+void MainWindow::navigateToCode(const QUrl &url, int lineNumber, int columnNumber)
+{
+    QSettings settings(QStringLiteral("KDAB"), QStringLiteral("Hotspot"));
+    settings.beginGroup(QStringLiteral("CodeNavigation"));
+    const auto ideIdx = settings.value(QStringLiteral("IDE"), -1).toInt();
+
+    QString command;
+#if !defined(Q_OS_WIN) && !defined(Q_OS_OSX) // Remove this #if branch when adding real data to ideSettings for Windows/OSX.
+    if (ideIdx >= 0 && ideIdx < ideSettingsSize) {
+        command += QString::fromUtf8(ideSettings[ideIdx].app);
+        command += QString::fromUtf8(" ");
+        command += QString::fromUtf8(ideSettings[ideIdx].args);
+    } else
+#endif
+    if (ideIdx == -1) {
+        command = settings.value(QStringLiteral("CustomCommand")).toString();
+    } else {
+        QDesktopServices::openUrl(QUrl(url));
```

Comment:
this should return early, as we open the file here. no need to open it below via qprocess a second time?